libraries/botbuilder-ai/src/qnaMakerDialog.ts

```diff
@@ -154,31 +148,104 @@ export class QnAMakerDialog extends WaterfallDialog implements QnAMakerDialogCon
 
+    /**
+     * The default threshold for answers returned, based on score.
+     */
+    protected defaultThreshold = 0.3;
+
+    /**
+     * The default maximum number of answers to be returned for the question.
+     */
+    protected defaultTopN = 3;
+
+    private currentQuery = 'currentQuery';
+    private qnAData = 'qnaData';
+    private defaultNoAnswer = 'No QnAMaker answers found.';
+
+    // Card parameters
+    private defaultCardTitle = 'Did you mean:';
+    private defaultCardNoMatchText = 'None of the above.';
+    private defaultCardNoMatchResponse = 'Thanks for the feedback.';
+
+    /**
+     * Gets or sets the QnA Maker knowledge base ID to query.
+     */
     public knowledgeBaseId: StringExpression;
+
+    /**
+     * Gets or sets the QnA Maker host URL for the knowledge base.
+     */
     public hostname: StringExpression;
+
+    /**
+     * Gets or sets the QnA Maker endpoint key to use to query the knowledge base.
+     */
     public endpointKey: StringExpression;
-    public threshold: NumberExpression = new NumberExpression(0.3);
-    public top: IntExpression = new IntExpression(3);
+
+    /**
+     * Gets or sets the threshold for answers returned, based on score.
+     */
+    public threshold: NumberExpression = new NumberExpression(this.defaultThreshold);
+
+    /**
+     * Gets or sets the maximum number of answers to return from the knowledge base.
+     */
+    public top: IntExpression = new IntExpression(this.defaultTopN);
+
+    /**
+     * Gets or sets the template to send to the user when QnA Maker does not find an answer.
+     */
     public noAnswer: TemplateInterface<Partial<Activity>, DialogStateManager> = new BindToActivity(
-        MessageFactory.text(this.defaultNoAnswer) as Activity
+        MessageFactory.text(this.defaultNoAnswer)
     );
+
+    /**
+     * Gets or sets the card title to use when showing active learning options to the user.
+     */
     public activeLearningCardTitle: StringExpression;
+
+    /**
+     * Gets or sets the button text to use with active learning options, allowing a user to
+     * indicate non of the options are applicable.
+     */
     public cardNoMatchText: StringExpression;
+
+    /**
+     * Gets or sets the template to send to the user if they select the no match option on an
+     * active learning card.
+     */
     public cardNoMatchResponse: TemplateInterface<Partial<Activity>, DialogStateManager> = new BindToActivity(
-        MessageFactory.text(this.defaultCardNoMatchResponse) as Activity
+        MessageFactory.text(this.defaultCardNoMatchResponse)
     );
+
+    /**
+     * Gets or sets the QnA Maker metadata with which to filter or boost queries to the knowledge base,
+     * or null to apply none.
+     */
     public strictFilters: ArrayExpression<QnAMakerMetadata>;
+
+    /**
+     * Gets or sets the flag to determine if personal information should be logged in telemetry.
+     *
+     * @remarks
```

Comment:
I think `@summary` is the proper JSDoc annotation here - `@remarks` always shows up as a lint error.